statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np 
+import pandas as pd 
+
+class SurveyDesign(object):
+
+    def __init__(self, strata=None, cluster=None, weights=None, nest=True):
+
+    
+        # if any of them are none, we should make them an array of ones
+        # how to know the len each array should be efficiently?
```

Comment:
I would have a function like below, then call it at the top of init.  It raises errors if needed and returns a clean set of arrays (all of ndarray type with consistent lengths):

```
def _check_args(self, strata, cluster, weights):
    if all([x is None for x in (strata, cluster, weights)]):
        raise ValueError("At least one of strata, cluster, and weights musts not be None")
    v = [len(x) if x is not None for x in (strata, cluster, weights)]
    if len(set(v)) != 1:
        raise ValueError("lengths of strata, cluster, and weights are not compatible")
    n = v[0]
    vals = []
    for x in (strata, cluster, weights):
        if x is None:
            vals.append(np.ones(n))
        else:
            vals.append(np.asarray(x))

    return vals[0], vals[1], vals[2] 
```